pkg/status/status.go

```diff
@@ -47,6 +47,9 @@ func Update(ctx context.Context, client k8sclient.Client, metallb *metallbv1beta
 }
 
 func getConditions(condition string, reason string, message string) []metav1.Condition {
+	if reason == "" {
```

Comment:
Where  are we calling this with reason="" ? If so, I think we should consider setting it from outside, to make the behaviour more clear

Reply:
It's from controller code  https://github.com/metallb/metallb-operator/blob/a36ae9fb1d40f10bc8d55b2ff772c4664f377acd/controllers/metallb_controller.go#L110-L116 and updated it